docs/source/markdown/guides/how_to/models/pre_processor.md

```diff
@@ -0,0 +1,250 @@
+```{eval-rst}
+:orphan:
+```
+
+# Pre-processing in Anomalib
+
+This guide explains how pre-processing works in Anomalib, its integration with models, and how to create custom pre-processors.
+
+## Overview
+
+Pre-processing in Anomalib is designed to encapsulate model-specific transforms (like input size and normalization) within the model graph itself. This design ensures that during deployment:
+
+- Pre-processing is part of the exported model (ONNX, OpenVINO)
+- Users don't need to manually resize or normalize inputs
+- Edge deployment is simplified with automatic pre-processing
+
+The `PreProcessor` class serves two roles:
+
+1. A PyTorch Module for transform application that gets exported with the model
+2. A Lightning Callback for managing stage-specific pre-processing steps (e.g, training, inference)
+
+For example, when a model is exported:
+
+```python
+# During training/development
+model = Patchcore(
+    pre_processor=PreProcessor(
+        transform=Compose([
+            Resize((256, 256)),
+            Normalize(mean=[0.485, 0.456, 0.406],
+                     std=[0.229, 0.224, 0.225])
+        ])
+    )
+)
+
+# Export model with pre-processing included
+model.export("model.onnx")
+
+# During deployment - no manual pre-processing needed
+deployed_model = onnxruntime.InferenceSession("model.onnx")
+raw_image = cv2.imread("test.jpg")  # Any size, unnormalized
+prediction = deployed_model.run(None, {"input": raw_image})
+```
+
+## Basic Usage
+
+```python
+from anomalib.pre_processing import PreProcessor
+from torchvision.transforms.v2 import Resize, Compose, ToTensor
+
+# Create a simple pre-processor
+pre_processor = PreProcessor(
+    transform=Compose([
+        Resize((256, 256)),
+        ToTensor()
+    ])
+)
+
+# Apply to a batch
+transformed_batch = pre_processor(batch)
+```
+
+## Stage-Specific Transforms
+
+You can specify different transforms for training, validation, and testing:
+
+```python
+from torchvision.transforms.v2 import RandomHorizontalFlip
+
+pre_processor = PreProcessor(
+    train_transform=Compose([
+        Resize((256, 256)),
+        RandomHorizontalFlip(),
+        ToTensor()
+    ]),
+    val_transform=Compose([
+        Resize((256, 256)),
+        ToTensor()
+    ]),
+    test_transform=Compose([
+        Resize((256, 256)),
+        ToTensor()
+    ])
+)
+```
+
+## Integration with Models
+
+The pre-processor is automatically integrated into Anomalib models through the `AnomalibModule` base class:
+
+```python
+from anomalib.models import Patchcore
+
+# Model creates default pre-processor
+model = Patchcore()
+
+# Or specify custom pre-processor
+model = Patchcore(
+    pre_processor=PreProcessor(
+        transform=Compose([
+            Resize((256, 256)),
+            ToTensor()
+        ])
+    )
+)
+
+print(model.pre_processor.train_transform)
+# Compose(
+#       Resize(size=[256, 256], interpolation=InterpolationMode.BILINEAR, antialias=False)
+#       ToTensor()
+# )
```

Comment:
- `train_transform` --> `transform` to reflect most recent design.
- removed `ToTensor` to avoid confusing users.

```suggestion
model = Patchcore(
    pre_processor=PreProcessor(
        transform=Resize((256, 256)),
    )
)

print(model.pre_processor.transform)
# Resize(size=[256, 256], interpolation=InterpolationMode.BILINEAR, antialias=False)
```